lib/puppet/face/catalog/pull.rb

```diff
@@ -64,7 +68,11 @@
     when_invoked do |catalog1, catalog2, args, options|
       require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'catalog-diff', 'searchfacts.rb'))
       require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'catalog-diff', 'compilecatalog.rb'))
-      nodes = Puppet::CatalogDiff::SearchFacts.new(args).find_nodes(options)
+      unless options[:node_list].nil?
+        nodes = options[:node_list].split(',')
+      else
+        nodes = Puppet::CatalogDiff::SearchFacts.new(args).find_nodes(options)
+      end
```

Comment:
Slightly more Rubyish way to write this (and also easier to understand `if`/`else` instead of `unless`/`else` b/c it’s not a double-negative):

```ruby
nodes = if options[:node_list].nil?
          Puppet::CatalogDiff::SearchFacts.new(args).find_nodes(options)
        else
          options[:node_list].split(',')
        end
```

Reply:
Good call!